src/test/java/com/mongodb/hibernate/jdbc/MongoResultSetTests.java

```diff
@@ -0,0 +1,170 @@
+            assertEquals("MongoResultSet has been closed", exception.getMessage());
+        }
+
+        private static Stream<Arguments> getMongoResultSetMethodInvocationsImpactedByClosing() {
+            return Map.<String, ResultSetMethodInvocation>ofEntries(
+                            Map.entry("next()", ResultSet::next),
+                            Map.entry("wasNull()", ResultSet::wasNull),
+                            Map.entry("getString(int)", rs -> rs.getString(1)),
+                            Map.entry("getBoolean(int)", rs -> rs.getBoolean(1)),
+                            Map.entry("getByte(int)", rs -> rs.getByte(1)),
+                            Map.entry("getShort(int)", rs -> rs.getShort(1)),
+                            Map.entry("getInt(int)", rs -> rs.getInt(1)),
+                            Map.entry("getLong(int)", rs -> rs.getLong(1)),
+                            Map.entry("getFloat(int)", rs -> rs.getFloat(1)),
+                            Map.entry("getDouble(int)", rs -> rs.getDouble(1)),
+                            Map.entry("getBytes(int)", rs -> rs.getBytes(1)),
+                            Map.entry("getDate(int)", rs -> rs.getDate(1)),
+                            Map.entry("getTime(int)", rs -> rs.getTime(1)),
+                            Map.entry("getTime(int,Calendar)", rs -> rs.getTime(1, Calendar.getInstance())),
+                            Map.entry("getTimestamp(int)", rs -> rs.getTimestamp(1)),
+                            Map.entry("getTimestamp(int,Calendar)", rs -> rs.getTimestamp(1, Calendar.getInstance())),
+                            Map.entry("getBigDecimal(int)", rs -> rs.getBigDecimal(1)),
+                            Map.entry("getArray(int)", rs -> rs.getArray(1)),
+                            Map.entry("getObject(int,Class)", rs -> rs.getObject(1, String.class)),
+                            Map.entry("getMetaData()", ResultSet::getMetaData),
+                            Map.entry("findColumn(String)", rs -> rs.findColumn("name")),
+                            Map.entry("isWrapperFor(Class)", rs -> rs.isWrapperFor(MongoResultSet.class)))
+                    .entrySet()
+                    .stream()
+                    .map(entry -> Arguments.of(entry.getKey(), entry.getValue()));
+        }
+    }
+
+    @Nested
+    class ColumnIndexCheckingTests {
+
+        @ParameterizedTest(
+                name = "SQLException is thrown when \"{0}\" is called on a MongoResultSet with columnIndex too low")
+        @MethodSource("getMongoResultSetMethodInvocationsWithColumnIndexUnderflow")
+        void testColumnIndexUnderflow(String label, ResultSetMethodInvocation methodInvocation) {
+            // when && then
+            var exception = assertThrows(SQLException.class, () -> methodInvocation.runOn(mongoResultSet));
+            assertTrue(exception.getMessage().startsWith("Invalid column index"));
+        }
+
+        @ParameterizedTest(
```

Comment:
Should we have a test that checks that calling it with a valid number works?

Reply:
I think if we cover the value verification testing cases as suggested above, we might not need to test calling it with a valid number works.

The tricky part here is it is easy to verify valid number won't trigger those column index related exceptions, but to avoid throwing other exceptions, lots of data preparing work needs to be done.

Reply:
It is probably best to eventually create a round trip test for each type of data. We could do this later (with a ticket).

Reply:
yeah, round trip testing seems fantastic idea. I'll do it soon.

Reply:
I added getting side testing for each type of data in the latest unit testing case, but I also found the necessity of creating round trip testing to ensure `PreparedStatement` and `ResultSet` work in tendom.

Reply:
Yes, nice test.